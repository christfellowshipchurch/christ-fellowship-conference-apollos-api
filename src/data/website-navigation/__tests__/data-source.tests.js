import ApollosConfig from '@apollosproject/config';
import WebsiteNavigation from '../data-source';

ApollosConfig.loadJs({
    ROCK: {
        API_URL: 'https://apollosrock.newspring.cc/api',
        API_TOKEN: 'some-rock-token',
        IMAGE_URL: 'https://apollosrock.newspring.cc/GetImage.ashx',
    },
    ROCK_MAPPINGS: {
        WEBSITE_CONTENT_CHANNEL_IDS: {
            TEST_SITE: [1]
        }
    }
})

const website = 'TEST_SITE'

describe('Website Navigation', () => {
    // Test getWebsiteNavigation
    it('requests the website navigation based on a website key', async () => {
        const dataSource = new WebsiteNavigation()

        dataSource.get = jest.fn(() => Promise.resolve([]))

        const result = await dataSource.getWebsiteNavigation(website)

        expect(dataSource.get.mock.calls).toMatchSnapshot()
    })

    it('requests the website navigation for a key that does not exist and returns null', async () => {
        const dataSource = new WebsiteNavigation()

        dataSource.get = jest.fn(() => Promise.resolve([]))

        const result = await dataSource.getWebsiteNavigation('INVALID_WEBSITE')

        expect(result).toBe(null)
    })
});