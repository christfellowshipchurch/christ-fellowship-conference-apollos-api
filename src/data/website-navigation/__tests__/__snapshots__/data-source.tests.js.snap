// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Website Navigation requests the website navigation based on a website key 1`] = `
Array [
  Array [
    "ContentChannels?loadAttributes=expanded&%24filter=Id%20eq%201&%24top=1",
    Object {},
    Object {},
  ],
]
`;
